flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/rules/AbstractNullAwareCodeGeneratorCastRule.java

```diff
@@ -59,8 +60,8 @@ public CastCodeBlock generateCodeBlock(
             LogicalType targetType) {
         final CastRuleUtils.CodeWriter writer = new CastRuleUtils.CodeWriter();
 
-        // Result of a casting can be null only and only if the input is null
-        final boolean isResultNullable = inputType.isNullable();
+        // Result of a casting is null if the input is null or the output primitive type is nullable
```

Comment:
nit: is this comment helpful? it basically describes the actually very readable expression below.